server/src/main/java/org/elasticsearch/rest/action/search/SearchCapabilities.java

```diff
@@ -22,9 +26,25 @@ private SearchCapabilities() {}
     private static final String RANGE_REGEX_INTERVAL_QUERY_CAPABILITY = "range_regexp_interval_queries";
     /** Support synthetic source with `bit` type in `dense_vector` field when `index` is set to `false`. */
     private static final String BIT_DENSE_VECTOR_SYNTHETIC_SOURCE_CAPABILITY = "bit_dense_vector_synthetic_source";
+    /** Support kql query. */
+    private static final String KQL_QUERY_SUPPORTED = "kql_query";
```

Comment:
I am not very happy to declare the capabilities here but:

- The `kql` plugin doesn't provide an endpoint that I could use, and `_search` seems like the most suitable endpoint to attach a query capability.
- There is no mechanism allowing a `SearchPlugin` to declare capacities to be added to the _search endpoint. It would be ideal that we have such a method but it is beyond the scope of this PR.

Do you folks have any suggestion to improve this or are you happy with this version?

Reply:
@afoucret I think this is ok, but does seem to indicate we don't have a nice way for plugins to declare capabilities on core APIs. @elastic/es-core-infra do we have a good way for plugins to declare capabilities on a server API?

Reply:
No, currently capabilities need to be declared on the rest handler that is responsible for that endpoint. There's no way to add additional capabilities later on.

Reply:
OK. Thank you all for the feedback.